===
layout: default
title: Testing for no change in RNA-seq differential gene expression
===

( from the bioconductor mailing list; Simon Anders)

DESeq2  provides a empirical- Bayesstyle shrinkage estimates of coefficients (i.e., log fold changes) andalso estimates standard errors for these coefficients (taken from thethe reciprocal curvature of the posterior). Your task is one of theapplication we had in mind for this.You want to know whether the fold change is zero or nearly zero, andyoualso want to be ascertained that this estimate of a nearly-zero foldchange is a precise one. So, to find genes whose absolute log foldchange is _reliably_ smaller than some threshold, take all those genesfor which the estimated abs log fold change is below the threshold andthe standard error is well below the threshold, too.The reason, why I write "below a threshold" rather than "equal tozero"is that it's biologically unreasonable to assume that a treatment hasexactly zero effect on a gene's expression. Gene regulation is such aninterconnected network that, at least in my view, every gene willreactever so slightly to every perturbance, but often, this reaction is toosmall to be measurable. This is why you should define a threshold for"biologically unlikely to be relevant". Let's say, we don't believethatan expression change of less than 15% (0.2 on a log2 scale) is worthbothering. So, if you might take all genes with abs log2 fold changebelow 0.2, and furthermore require that the standard error of thislog2fold change estimate is below, say, 0.05, than you will get gene withtrue values well below at least 0.3 or so.In the end, the thresholds won't matter too much but if you want realpvalues, this is possible, too: Simply divide the log2 fold changeestimates by their standard errors to get z scores (this works becausethe sampling distribution of our fold change estimates is reasonablyclose to normal), and do two one-sided tests (TOST) as suggested byRyanabove.
