<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title></title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="//js/jquery.min.js"></script>
    <script src="//js/bootstrap.min.js"></script>
    <link href="//css/bootstrap.min.css" rel="stylesheet">
    <link href="//css/theme.css" rel="stylesheet">
</head>

<body>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                  <span class="sr-only">Toggle navigation</span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="//"></a>
              </div>
              <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li class="active"><a href="//">Home</a></li>
                    <li class="active visible-xs-block"><a href="//links.html">Links</a></li>
                    <li class="active"><a href="//archive.html">Archive</a></li>
                    <li class="active"><a href="//about.html">About</a></li>
                    <li class="active"><a href="//feed.xml">RSS</a></li>
                    
                </ul>
              </div>
        </div>
    </div>
</div>


<div class="container container-left">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class="sidebar well">
Genomics data scientist
</div>

<div class="sidebar well">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="//blog/lorne2012poster.html">Lorne Genome 2012 poster</a></li>
        
          <li><a href="//blog/ciRNApapers.html">2013 ciRNA papers</a></li>
        
          <li><a href="//blog/helloworld.html">Hello World!</a></li>
        
    </ul>
</div>

<div class="sidebar well">
<h1>Links</h1>
<ul>
  <li><a href="#">One</a></li>
  <li><a href="#">Two</a></li>
  <li><a href="#">Three</a></li>
  <li><a href="#">Four</a></li>
</ul>

</div>

        </div>
        <div class="col-md-9">
            <h2 id="sequence-with-biomart">Sequence with biomart</h2>
<div class="highlight"><pre><code class="language-text" data-lang="text">getSequence {biomaRt}   R Documentation

Retrieves sequences

Description

This function retrieves sequences given the chomosome, start and end position or a list of identifiers. Using getSequence in web service mode (default) generates 5&#39; to 3&#39; sequences of the requested type on the correct strand. The type of sequence returned can be specified by the seqType argument which takes the following values: &#39;cdna&#39;;&#39;peptide&#39; for protein sequences;&#39;3utr&#39; for 3&#39; UTR sequences,&#39;5utr&#39; for 5&#39; UTR sequences; &#39;gene_exon&#39; for exon sequences only; &#39;transcript_exon_intron&#39; gives the full unspliced transcript, that is exons + introns;&#39;gene_exon_intron&#39; gives the exons + introns of a gene;&#39;coding&#39; gives the coding sequence only;&#39;coding_transcript_flank&#39; gives the flanking region of the transcript including the UTRs, this must be accompanied with a given value for the upstream or downstream attribute;&#39;coding_gene_flank&#39; gives the flanking region of the gene including the UTRs, this must be accompanied with a given value for the upstream or downstream attribute; &#39;transcript_flank&#39; gives the flanking region of the transcript exculding the UTRs, this must be accompanied with a given value for the upstream or downstream attribute; &#39;gene_flank&#39; gives the flanking region of the gene excluding the UTRs, this must be accompanied with a given value for the upstream or downstream attribute. In MySQL mode the getSequence function is more limited and the sequence that is returned is the 5&#39; to 3&#39;+ strand of the genomic sequence, given a chromosome, as start and an end position. So if the sequence of interest is the minus strand, one has to compute the reverse complement of the retrieved sequence, which can be done using functions provided in the matchprobes package. The biomaRt vignette contains more examples on how to use this function.




Usage

getSequence( chromosome, start, end, id, type, seqType, upstream, downstream, mart, verbose=FALSE)
Arguments

chromosome   Chromosome name
start    start position of sequence on chromosome
end  end position of sequence on chromosome
id   An identifier or vector of identifiers.
type     The type of identifier used. Supported types are hugo, ensembl, embl, entrezgene, refseq, ensemblTrans and unigene. Alternatively one can also use a filter to specify the type. Possible filters are given by the listFilters function
seqType  Type of sequence that you want to retrieve. Allowed seqTypes are: cdna, peptide, 3utr, 5utr, genomic
upstream     To add the upstream sequence of a specified number of basepairs to the output.
downstream   To add the downstream sequence of a specified number of basepairs to the output.
mart     object of class Mart created using the useMart function
verbose  If verbose = TRUE then the XML query that was send to the webservice will be displayed.
</code></pre></div>
<p>Examples</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">if(interactive()){
mart &lt;- useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)

seq = getSequence(id=&quot;BRCA1&quot;, type=&quot;hgnc_symbol&quot;, seqType=&quot;peptide&quot;, mart = mart)
show(seq)

seq = getSequence(id=&quot;1939_at&quot;, type=&quot;affy_hg_u95av2&quot;, seqType=&quot;gene_flank&quot;,upstream = 20, mart = mart)
show(seq)

}

seq = getSequence(id=U1all$ensembl_gene_id, type=&quot;ensembl_gene_id&quot;, seqType=&quot;cdna&quot;, mart = ensembl)
write.fasta(sequences=as.list(seq$cdna),names=seq$ensembl_gene_id,file.out=&quot;write.my.dna.fasta&quot;)
</code></pre></div>
        </div>
    </div>
</div>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2015 . Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>






</body>
</html>

