<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title></title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="//js/jquery.min.js"></script>
    <script src="//js/bootstrap.min.js"></script>
    <link href="//css/bootstrap.min.css" rel="stylesheet">
    <link href="//css/theme.css" rel="stylesheet">
</head>

<body>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                  <span class="sr-only">Toggle navigation</span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                  <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="//"></a>
              </div>
              <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li class="active"><a href="//">Home</a></li>
                    <li class="active visible-xs-block"><a href="//links.html">Links</a></li>
                    <li class="active"><a href="//archive.html">Archive</a></li>
                    <li class="active"><a href="//about.html">About</a></li>
                    <li class="active"><a href="//feed.xml">RSS</a></li>
                    
                </ul>
              </div>
        </div>
    </div>
</div>


<div class="container container-left">
    <div class="row">
        <div class="col-md-3 hidden-xs">
            <div class="sidebar well">
Genomics data scientist
</div>

<div class="sidebar well">
    <h1>Recent Posts</h1>
    <ul>
        
          <li><a href="//blog/lorne2012poster.html">Lorne Genome 2012 poster</a></li>
        
          <li><a href="//blog/ciRNApapers.html">2013 ciRNA papers</a></li>
        
          <li><a href="//blog/helloworld.html">Hello World!</a></li>
        
    </ul>
</div>

<div class="sidebar well">
<h1>Links</h1>
<ul>
  <li><a href="#">One</a></li>
  <li><a href="#">Two</a></li>
  <li><a href="#">Three</a></li>
  <li><a href="#">Four</a></li>
</ul>

</div>

        </div>
        <div class="col-md-9">
            <h2 id="interpreting-output-logs">Interpreting output logs</h2>

<p>Example taken from <a href="http://seqanswers.com/forums/showthread.php?t=29769">here</a></p>
<div class="highlight"><pre><code class="language-text" data-lang="text">Started job on |    Mar 15 04:59:21
Started mapping on |    Mar 15 05:02:08
Finished on |   Mar 15 05:29:49
Mapping speed, Million of reads per hour |  99.16

Number of input reads | 45750724
Average input read length | 202
UNIQUE READS:
Uniquely mapped reads number |  3331104
Uniquely mapped reads % |   7.28%
Average mapped length | 199.28
Number of splices: Total |  736416
Number of splices: Annotated (sjdb) |   665151
Number of splices: GT/AG |  733030
Number of splices: GC/AG |  2159
Number of splices: AT/AC |  569
Number of splices: Non-canonical |  658
Mismatch rate per base, % | 1.11%
Deletion rate per base |    0.04%
Deletion average length |   2.30
Insertion rate per base |   0.03%
Insertion average length |  2.02
MULTI-MAPPING READS:
Number of reads mapped to multiple loci |   578140
% of reads mapped to multiple loci |    1.26%
Number of reads mapped to too many loci |   43288
% of reads mapped to too many loci |    0.09%
UNMAPPED READS:
% of reads unmapped: too many mismatches |  0.00%
% of reads unmapped: too short |    1.19%
% of reads unmapped: other |    90.17%
</code></pre></div>
<p>Alex Dobin&#39;s response</p>

<p>First check the uniquely mapped reads:</p>

<p>Average mapped length | 199.28 : good, close you your pair length of 202</p>

<p>Mismatch rate per base, % | 1.11% : a bit on the high side, you would get 0.5-0.8% for good libraries, </p>

<p>The splices are dominated by annotated and canonical, which is good.</p>

<p>The indel rate is low.</p>

<p>So, the reads that actually mapped uniquely - as few as they are - look fine.</p>

<p>The ratio of unique to multimappers is 7.28%/1.26% ~ 6 is somewhat high, that is - for typical human cells, </p>

<p>I am not sure what are you sequencing. Our typical value is 15-20.</p>

<p>% of reads mapped to too many loci |    0.09% : by default &quot;too many loci&quot; is &gt;10, but this number is good so you are not missing much.</p>

<p>Finally - most importantly - unmapped reads.</p>

<p>% of reads unmapped: too short |    1.19% : this number would be large if you had poor sequencing quality, </p>

<p>it is surprisingly small (we typically get ~5%).</p>

<p>% of reads unmapped: other |    90.17% : </p>

<p>this where all the unmapped reads went and it is very unusual.</p>

<p>It means that for 90% of the reads STAR could not find good anchor seeds. Two main possibilities are:</p>

<ol>
<li>Contamination. Most reads have very little homology with human genome. </li>
</ol>

<p>You can check it by BLASTing a few unmapped reads against everything.</p>

<ol>
<li>Repeat regions dominate expression. The number of loci a seed could map to is limited by </li>
</ol>

<p>--winAnchorMultimapNmax = 50 by default. You could increase it to ~1000 to see if more reads get mapped</p>

<p>(also increase --outFilterMultimapNmax to output them as multi-mappers).</p>

<p>Other comments:</p>

<p>We typically see higher than 80% mapping rate for our RNA-Seq differential expression projects as well</p>

<p>ribosomal RNA contamination of mRNA-Seq libraries can produce this type of result due to the high copy number of the rRNA clusters. Adapter dimers are another possible culprit.</p>

<p>If the number of unmapped reads &quot;too short&quot; is very high</p>

<p>There are three main explanations for poor mappability with large proportion of unmapped alignments reported as &quot;too short&quot;: </p>

<p>From <a href="https://groups.google.com/forum/#!topic/rna-star/7RwKkvNLmI4%20thread">STAR google groups</a></p>

<ol>
<li>Poor sequencing quality.</li>
</ol>

<p>On your quality scores distribution plots, If the median quality score drops below 20 at a certain cycle, you will have problems mapping the tails.
Note that by default STAR requires mapped length to be &gt; 2/3 of the total read length (i.e. 2/3*202=135b in case of PE101).
This is controlled by --outFilterMatchNminOverLread and --outFilterScoreMinOverLread.
You can try to reduce this parameters to, say, 0.4 to see if you get more reads mapped - but the alignments will be shorter, of course.
Also, you can try to map read1 and read2 separately to see if one of them is more problematic than the other.</p>

<p>&#39;&#39;&#39;2a. Contamination with exogenous sequences.&#39;&#39;&#39;</p>

<p>You can try to BLAST a few of unmapped reads agains the full NCBI database to see if you get any good matches.</p>

<p>&#39;&#39;&#39;2b. Contamination with ribosomal RNA.&#39;&#39;&#39;</p>

<p>If your samples are &quot;total RNA&quot;, depleted with Ribo-Zero or Ribo-Minus kits, it is possible that the depletion did not work well. rRNA are typically multi-mappers (and you get plenty of those), however, not all rRNA repeats make it into the main chromosomal assembly, and in this case they will not be mapped and will be reported as &quot;alignment too short&quot;. We have recently had many cases like that in our lab for human tissues. I believe for the fly genome, the unplaced contigs are in chrU and chrUextra - please try to include them in the genome if you have not done so.</p>

<p>&#39;&#39;&#39;2c. Contamination with primer-dimers.&#39;&#39;&#39;</p>

<p>You can try to clip off Illumina adapters with various clipper software. I think this is quite rare.</p>

<p>&#39;&#39;&#39;3. Inserts that are too short.&#39;&#39;&#39;</p>

<p>If this happens, you will be sequencing into your adapter at the end of the 2nd mate. STAR will try to trim it off, but  the resulting alignment might be too short.
You can check if this is the case by following the suggestions in 1 or 3c.</p>

<p>By default STAR extends alignments until the number of mismatches exceeds --outFilterMismatchNmax (=10 by default) or the number of mismatches divided by read length exceeds --outFilterMismatchNoverLmax (=0.3 default). If you have ~10% sequence divergence, for ~200b reads you would expect ~20 mismatches per read, so the reads will be cut short because of the first parameter. Please try to increase it to a larger number - I would make it 100 and control the allowed mismatch rate with  --outFilterMismatchNoverLmax - if you average mismatch rate is 10%, the default --outFilterMismatchNoverLmax 0.3 is probably a good choice.</p>

<p>To increase sensitivity for such a high mismatch rate, it&#39;s also useful to reduce --seedSearchStartLmax (=50 by default), I would try 25 or even lower.</p>

<p>Working with very short reads (25 nt, as for example from ribosomal profiling)</p>

<p>25b reads will be much harder to map to the splice junctions, of course. Using annotations at the genome generation step would be very beneficial, with --sjdbOverhang chosen to be the longest read length -1.</p>

<p>At the mapping step, you can increase sensitivity with --seedSearchStartLmax 20, or even 15.</p>

<p>If you are using SJ.out.tab file, I would reduce --outSJfilterOverhangMin from 30 12 12 12 to, say, 30  8 8 8, to increase sensitivity for unannotated junctions.</p>

<p>You may want to be more restrictive for minimum allowed mapped length/score, by increasing:
outFilterScoreMin               0</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">int: alignment will be output only if its score is higher than this value
</code></pre></div>
<p>outFilterScoreMinOverLread      0.66</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">    float: outFilterScoreMin normalized to read length (sum of mates&#39; lengths for paired-end reads)
</code></pre></div>
<p>outFilterMatchNmin              0
    int: alignment will be output only if the number of matched bases is higher than this value</p>

<p>outFilterMatchNminOverLread     0.66</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">float: outFilterMatchNmin normalized to read length (sum of mates&#39; lengths for paired-end reads)
</code></pre></div>
<p>You can trim 3&#39; adapters from within  STAR with</p>

<p>clip3pAdapterSeq            -
    string(s): adapter sequences to clip from 3p of each mate.  If one value is given, it will be assumed the same for both mates.</p>

<p>clip3pAdapterMMp            0.1
    double(s): max proportion of mismatches for 3p adpater clipping for each mate.  If one value is given, it will be assumed the same for both mates.</p>

<p>Mapping to transcriptome only</p>

<p>Set the parameters
--alignIntronMax smaller than --alignIntronMin.
--alignIntronMax 0 has a special meaning and sets the max intron to be defined by window sizes, by default ~600kb
Choosing the correct value for Sjdboverhang Option</p>

<p>sjdbOverhang should be set as readlength -1</p>

<p>The &quot;Overhang&quot; in sjdbOverhang and  --alignSJDBoverhangMin has different meanings - bad naming choice, unfortunately.</p>

<p>The --sjdbOverhang is used only at the genome generation step, and tells STAR how many bases to concatenate from donor and acceptor sides of the junctions. If you have 100b reads, the ideal value of --sjdbOverhang is 99, which allows the 100b read to map 99b on one side, 1b on the other side. One can think of --sjdbOverhang as the maximum possible overhang for your reads.</p>

<p>On the other hand, --alignSJDBoverhangMin is used at the mapping step to define the minimum allowed overhang over splice junctions. For example, the default value of 3 would prohibit overhangs of 1b or 2b.</p>

<hr>

<h2 id="what-some-of-the-output-tags-in-the-bam-file-mean">What some of the output tags in the bam file mean</h2>

<p>From STAR google group: <a href="https://groups.google.com/forum/#!topic/rna-star/dYI_O-9hDXc">here</a>:</p>

<p>In STAR&#39;s output bam file there are lines with uT tags - unmapped type -, which are used for labeling unmapped reads:</p>

<ul>
<li>uT:A:0 - other (no good anchor seeds)</li>
<li>uT:A:1 - too short</li>
<li>uT:A:2 - too many mismatches</li>
<li>uT:A:3 - mapped to too many loci</li>
</ul>

        </div>
    </div>
</div>

<div class="container-fluid">
    <div class="row-fluid">
        <div class="span12 footer navbar-inverse navbar-fixed-bottom">
            <p class="copyright">&copy;2015 . Powered by <a href="http://jekyllrb.com">Jekyll</a>, theme by <a href="https://github.com/scotte/jekyll-clean">Scott Emmons</a>
            under
            <a href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution</a></p>
        </div>
    </div>
</div>






</body>
</html>

